appvia/kev
# Re-implement `render` UI using terminal UI components (based on PTerm.sh)

Resolves #433 

**Pending**

- [x] Complete Rendering manifests UI.
- [x] Adding error UI sections.
- [x] Updating tests.

## Sample Output

```shell
$ kev render -e prod

» Loading...

» Validating compose sources...
Detecting secrets in: docker-compose.yaml
 !  Detected in service: db-test
   | env var [MYSQL_USER] - Contains word: user
   | env var [MYSQL_PASSWORD] - Contains word: password
   | env var [MYSQL_ROOT_PASSWORD] - Contains word: password
 !  Detected in service: wordpress
   | env var [WORDPRESS_DB_HOST] - Contains word: host
   | env var [WORDPRESS_DB_USER] - Contains word: user
   | env var [WORDPRESS_DB_PASSWORD] - Contains word: password

Validation successful!
However, to prevent secrets leaking, see help page:
https://github.com/appvia/kev/blob/master/docs/reference/config-params.md#reference-k8s-secret-key-value

» Verifying Skaffold...
 ✓  Skaffold manifest is available

» Validating deployment environments...
Detecting secrets in: docker-compose.kev.prod.yaml
 ✓  Non detected in service: db-test
 ✓  Non detected in service: wordpress

Validation successful!

» Detecting project updates...
prod: docker-compose.kev.prod.yaml
 ✓  No version update detected
 ✓  No service additions detected
 ✓  No service removals detected
 ✓  No env var removals detected
 ✓  No volume additions detected
 ✓  No volume removals detected

» Rendering manifests, format: kubernetes...
prod: docker-compose.kev.prod.yaml
 ✓  Converted service: db-test
   | rendered StatefulSet
   | rendered Service
   | rendered PersistentVolumeClaim
 ✓  Converted service: wordpress
   | rendered Deployment
   | rendered Service
 ✓  Created networking policies
   | rendered NetworkPolicy

Project manifests rendered!
A set of 'kubernetes' manifests have been generated:
  prod: k8s/prod

The project can now be deployed to a Kubernetes cluster.

To test locally:
 - Ensure you have a local cluster up and running with a configured context.
 - Create a namespace: `kubectl create ns ns-example`.
 - Apply the manifests to the cluster: `kubectl apply -f <manifests-dir>/<env> -n ns-example`.
 - Discover the main service: `kubectl get svc -n ns-example`.
 - Port forward to the main service: `kubectl port-forward service/<service_name> <service_port>:<destination_port> -n ns-example`.
```
